CIP-0000.md

```diff
@@ -0,0 +1,198 @@
+- - - -
+lang: en
+cip: 0000
+title: Celo Improvement Proposal (CIP) Process
+status: Draft
+type: Meta
+discussions-to:
+author: Yaz Khoury (@YazzyYaz), James Prestwich (@prestwich), Pranay Mohan (@pranaymohan), Mariano Cortesi (@mcortesi)
+created: 2020-09-22
+license: CC0 1.0
+- - - -
+
+# Abstract
+A Celo Improvement Proposal (CIP) describe standards for the Celo platform, including the core protocol specifications, SDK, and contract standards. It is a design document providing information to the Celo community, or describing a new feature for Celo or its processes or environment, a rationale for the proposal, detailed solution including technical specifications, and, if any, a list of potential risks. 
+
+The CIP proposer, the **author** is responsible for building consensus within the community and documenting whether there is rough consensus on their proposal. 
+
+Because the CIPs are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal.
+
+Participation in the Celo project and the CIP process is informed by the Celo [Community Tenets](https://celo.org/community) and subject to the  [Code of Conduct](https://celo.org/code-of-conduct).
+
+# CIP Types
+There are three types of CIP:
+
+* A **Standards Track CIP** describes any change that affects most or all Celo implementations, such as a change to the network protocol, a change in block or transaction validity rules, proposed application standards/conventions, or any change or addition that affects the interoperability of applications using Celo. Furthermore Standard CIPs can be broken down into the following categories. Standards Track CIPs consist of two parts, a design document and implementation.
+	* **Core** - improvements requiring a consensus fork, as well as changes that are not necessarily consensus critical but may be relevant to  “core dev” discussions, and the validator/node strategy.
+	* **Governance** - improvements to the governance proposal to the Celo platform including parameter changes. It should explain the reason for the change, enumerate the smart contract methods that will be called with parameters, and describe the impact and any risks. 
```

Comment:
For a Celo on-chain governance proposal, how does this process interact with the formal on-chain process of upvotes, approvals and actual voting? I.e does the equivalent CIP first reach `Final` stage before it should be voted on, or can that already happen in `Draft`?

Reply:
It's a good question, I'm not 100% on what to do about Governance here. I was also thinking maybe it makes sense to move Governance out of it the equation and back to the CGP side.

The one thing I think we want to achieve with CIP process is to target things that need to be standardized, which is easy to follow from the EIP/BIP/PEP process since it's concerned with maintaining and upgrading a standard.

For Governance, we probably need to have a discussion about whether it makes sense to include it in the CIP process the more I think about it.

Reply:
Yeah I can definitely see the case for governance to be outside of the CIP process, IMO it's mostly an implementation detail to possibly achieve some standard, but it's likely never the standard itself. The one nice thing about governance is that it's currently the only process that receives the vote/approval of the Celo community as a whole

Reply:
Generally, the EIPs and BIPs are marked `Final` when they are activated. I expect a governance proposal would be marked `Final` once it has taken affect

Reply:
> The one nice thing about governance is that it's currently the only process that receives the vote/approval of the Celo community as a whole

This is the thing that is scariest about the process, not what's nice about it :thinking: